chainer/utils/type_check.py

```diff
@@ -583,3 +585,33 @@ def prod(xs):
         return _prod_impl(xs)
     else:
         return _prod(xs)
+
+
+def expect_broadcast_shapes(*shape_types, **kwargs):
+    """Checks if two shapes are broadcastable at first some axes.
+
+    Args:
+        shapes_types: Type-checked shapes of the arrays to broadcast.
+        ignore_tail (int): Number of axes at the tail to ignore.
```

Comment:
Should the function check the shapes have tail dimensions?  I prefer `type_check.expect_broadcast_shapes(a_type.shape[:-2], b_type.shape[:-2])` as the meaning of `ignore_tail` is not clear.

Reply:
Looks better. Fixed accordingly.